merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/AutoValueMappers.java

```diff
@@ -153,6 +188,123 @@ record ComponentMapperNamePair(String componentName, String mapperName) {}
               .build());
     }
 
+    record Property(String name, TypeMirror type, String mapperName) {}
+    for (final var annotation : annotationTypes) {
+      final var methodName = ClassName.get((TypeElement) annotation).canonicalName().replace(".", "_");
+      final var properties = new ArrayList<Property>();
+      final var necessaryMappers = new HashMap<TypeMirror, String>();
+      for (final var element : annotation.getEnclosedElements()) {
+        if (!(element.getKind() == ElementKind.METHOD)) continue;
+        final var typeMirror = ((ExecutableElement) element).getReturnType();
+        final var elementName = element.toString().replace("(", "").replace(")", "");
+        final var valueMapperIdentifier = JAVA_IDENTIFIER_ILLEGAL_CHARACTERS
+                                              .matcher(typeMirror.toString())
+                                              .replaceAll("_")
+                                          + "ValueMapper";
```

Comment:
Is it possible to get illegal characters in this `toString`? I don't mind doing the regex match and replace for safety, I'm just not clear if how it would occur here, since `typeMirror` is a Java return type.

Reply:
What we're trying to do here is turn a string representation of a type into a valid identifier. Unfortunately, types often contain characters that are illegal as Java identifiers - e.g. in `List<Double>`, `<` and `>` are illegal characters for identifiers.